### PR TITLE
EOS-18893 Replication idempotency

### PR DESCRIPTION
**Pull request by abhilekh268**

### CODE REVIEW

File: scripts/ldap/replication/setupReplicationScript.sh [hunk 1]
```diff
@@ -76,16 +76,31 @@ getServerIdFromHostFile()
     done < "$host_list"
 }
 
+op=$(ldapsearch -w seagate -x -D cn=admin,cn=config -b cn=config "olcServerID")
```
Comment (amitwac2608) [inline scripts/ldap/replication/setupReplicationScript.sh:79]: it seems the hardcode rootdn password being used here. Please ask for the rootdn password on cmdline.

File: scripts/ldap/replication/setupReplicationScript.sh [hunk 2]
```diff
@@ -76,16 +76,31 @@ getServerIdFromHostFile()
...
 
 sed -e "s/\${serverid}/$id/" $INSTALLDIR/serverIdTemplate.ldif > scriptServerId.ldif
 ldapmodify -Y EXTERNAL  -H ldapi:/// -f scriptServerId.ldif
 rm scriptServerId.ldif
 
-ldapadd -Y EXTERNAL -H ldapi:/// -f $INSTALLDIR/syncprov_mod.ldif
+ldapadd -Y EXTERNAL -H ldapi:/// -f $INSTALLDIR/syncprov_mod.ldif || echo "true"
 ldapadd -Y EXTERNAL -H ldapi:/// -f $INSTALLDIR/syncprov.ldif
 
+op=$(ldapsearch -w seagate -x -D cn=admin,cn=config -b olcDatabase={0}config,cn=config "olcSyncrepl")
```
Comment (amitwac2608) [inline scripts/ldap/replication/setupReplicationScript.sh:94]: do not use hardcoded rootdn password.

File: scripts/ldap/replication/setupReplicationScript.sh [hunk 3]
```diff
@@ -76,16 +76,31 @@ getServerIdFromHostFile()
...
+if [[ $op == *"olcSyncrepl:"* ]];then
+  ldapmodify -Y EXTERNAL  -H ldapi:/// -f cleanup/config.ldif
+fi
+
+op=$(ldapsearch -w seagate -x -D cn=admin,cn=config -b olcDatabase={0}config,cn=config "olcMirrorMode")
```
Comment (amitwac2608) [inline scripts/ldap/replication/setupReplicationScript.sh:99]: remove hard-coded rootdn password usage.

**Review by amitwac2608 — CHANGES_REQUESTED** [review]

**Review by sachitanands — APPROVED** [review]
pls address Amit's comment ..rest looks good to me.

File: scripts/ldap/replication/setupReplicationScript.sh [hunk 4]
```diff
@@ -76,16 +80,31 @@ getServerIdFromHostFile()
...
+
 # update serverID
 getServerIdFromHostFile
 
 sed -e "s/\${serverid}/$id/" $INSTALLDIR/serverIdTemplate.ldif > scriptServerId.ldif
 ldapmodify -Y EXTERNAL  -H ldapi:/// -f scriptServerId.ldif
 rm scriptServerId.ldif
 
-ldapadd -Y EXTERNAL -H ldapi:/// -f $INSTALLDIR/syncprov_mod.ldif
+ldapadd -Y EXTERNAL -H ldapi:/// -f $INSTALLDIR/syncprov_mod.ldif || echo "true"
```
Comment (shalakadharap) [inline scripts/ldap/replication/setupReplicationScript.sh:95]: Why is this echo statement?

Comment (shalakadharap) [reply scripts/ldap/replication/setupReplicationScript.sh:95]: Just 1  question, rest all looking good to me.

**Review by AjinkyaDhumal — APPROVED** [review]

File: scripts/ldap/replication/setupReplicationScript.sh [hunk 5]
```diff
@@ -103,6 +119,16 @@ echo "olcMirrorMode: TRUE" >> scriptConfig.ldif
...
 rm scriptConfig.ldif
 
+op=$(ldapsearch -w $password -x -D cn=admin,cn=config -b olcDatabase={2}mdb,cn=config "olcSyncrepl")
+if [[ $op == *"olcSyncrepl:"* ]];then
+  ldapmodify -Y EXTERNAL  -H ldapi:/// -f cleanup/data.ldif
+fi
+
+op=$(ldapsearch -w $password -x -D cn=admin,cn=config -b olcDatabase={2}mdb,cn=config "olcMirrorMode")
+if [[ $op == *"olcMirrorMode:"* ]];then
+  ldapmodify -Y EXTERNAL  -H ldapi:/// -f cleanup/olcmirromode_data.ldif
```
Comment (bkirunge7) [inline scripts/ldap/replication/setupReplicationScript.sh:129]: can you log success failure of these commands also print ldap commands at info level

File: scripts/ldap/replication/setupReplicationScript.sh [hunk 6]
```diff
@@ -103,6 +119,16 @@ echo "olcMirrorMode: TRUE" >> scriptConfig.ldif
 ldapmodify -Y EXTERNAL  -H ldapi:/// -f scriptConfig.ldif
 rm scriptConfig.ldif
 
+op=$(ldapsearch -w $password -x -D cn=admin,cn=config -b olcDatabase={2}mdb,cn=config "olcSyncrepl")
```
[the remark this answers is not in the document]
Comment (abhilekh268) [reply scripts/ldap/replication/setupReplicationScript.sh:124]: no,it gives all multivalued values.

File: scripts/ldap/replication/setupReplicationScript.sh (hunk 5 above)
Comment (abhilekh268) [reply scripts/ldap/replication/setupReplicationScript.sh:132]: this is bash script added echo statements.info level ldap commands ?

**Review by amitwac2608 — APPROVED** [review]

**Review by bkirunge7 — APPROVED** [review]

**Pull request merged**